sqlglot/dialects/snowflake.py

```diff
@@ -561,7 +561,20 @@ def _annotate_dateadd(self: TypeAnnotator, expression: exp.DateAdd) -> exp.DateA
         self._set_type(expression, exp.DataType.Type.TIMESTAMPNTZ)
     else:
         self._annotate_by_args(expression, "this")
+    return expression
+
+
+def _annotate_timeadd(self: TypeAnnotator, expression: exp.TimeAdd) -> exp.TimeAdd:
+    self._annotate_args(expression)
 
+    if expression.this.is_type(exp.DataType.Type.DATE) and expression.text("unit").upper() in (
+        "HOUR",
+        "MINUTE",
+        "SECOND",
```

Comment:
@fivetran-BradfordPaskewitz I think we should do something similar to your PR here right? The one that handled `dateadd`, L554-L564.